pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -70,7 +69,11 @@ func NewScheduler(ctx context.Context, kubeClient client.Client, nodeTemplates [
 	}
 
 	// wait to ensure that our cluster state is synced with the current known nodes to prevent over-shooting
-	s.waitForClusterStateSync(ctx)
+	if WaitForClusterSync {
+		for err := s.clusterStateSynchronized(ctx); err != nil; time.Sleep(1 * time.Second) {
```

Comment:
If this ever fails to sync, it infinite loops due to not calling `clusterStateSynchronized` again.